connectivity/FEATURE_BLE/include/ble/gap/Events.h

```diff
@@ -629,11 +673,22 @@ struct AdvertisingEndEvent {
         return connected;
     }
 
+    /** Is the end of legacy advertising.
+     *
+     * If it is the return of getConnection() getCompleted_events() and isConnectrd()
+     * must be discarded
+     */
+    bool isLegacy() const
+    {
+        return legacy;
+    }
+
 private:
     advertising_handle_t advHandle;
     connection_handle_t connection;
     uint8_t completed_events;
     bool connected;
+    bool legacy;
```

Comment:
I don't think you need an extra member. if you really need to know if it's legacy you can just check the handle

Reply:
The legacy handle is the only handle usable if extended advertising is not supported. However it doesn't mean it isn't usable with extended advertising. The handle alone isn't enough to indicate if legacy or extended advertising is being used.